composition/proposal.md

```diff
@@ -530,9 +530,90 @@ It would also generally be *not* suggested to force a static model to be
 non-static, as the static model may not be designed for non-static use, and
 waste time on debugging.
 
-#### 1.5 Minimal Interface Types for Non-SDFormat Models
+#### 1.5 Minimal `libsdformat` Interface Types for Non-SDFormat Models
+
+As mentioned above, the encapsulation goal of this proposal should allow for
+downstream libraries to permit specifying non-SDFormat model via `//include`
+tags *without* `libsdformat` having to try and convert the model to SDFormat.
+
+In order to do so, it is proposed that the following API hook be permitted to
+be registered in `libsdformat`:
+
+~~~c++
+struct sdf::NestedInclude {
+  /// Provides the *unresolved* file path as provided from the file directly
+  /// included.
+  std::string file_path;
+
+  /// Name of the model in absolute hierarchy.
+  /// Example: `top_model::middle_model::my_new_model`
+  // N.B. Should be unnecesssary if downstream consumer has composition. Not
+  // the case for Drake :(
+  std::string absolute_model_name;
+
+  /// Name relative to immediate parent.
+  /// Example: `my_new_model`
+  std::string local_model_name;
+
+  /// Pose information from //include/pose.
+  sdf::SemanticPose pose;
+};
+
+class sdf::InterfaceModel {
+  /// \param[in] name The *local* name (no nesting, e.g. "::").
+  public: InterfaceModel(std::string name);
+  /// Provided so that hierarchy can still be leveraged from SDFormat.
+  public: void AddNestedModel(sdf::InterfaceModelPtr nested_model);
+  /// Provided so that the including SDFormat model can still interface with
+  /// the declared frames.
+  public: void AddFrame(sdf::InterfaceFramePtr frame);
+};
+
+class sdf:::InterfaceFrame {
+  /// \param[in] name The *local* name.
+  /// \param[in] pose The semantic pose of the frame.
+  ///   If the parent frame is specified as "", then the frame itself is
+  ///   considered as a link, and can be used in `attached_to` to denote an
+  ///   anchoring in the frame graph.
+  public: InterfaceFrame(std::string name, sdf::SemanticPosePtr pose);
+};
+
+/// \param[in] include The //include tag information from which this model
+///   should be parsed.
+/// \param[out] errors Errors encountered during custom parsing.
+/// \returns An optional ModelInterface.
+///   * If not nullptr, hte returned model interface is incorprated into the
+///     existing model and its frames are exposed through the frame graph.
+///   * If nullptr, then libsdformat should continue testing out other custom
+///     parsers registered under the same extension (e.g. a parser for
+///     `.yaml`, which may cover many different schemas).
+using sdf::CustomModelParser = std::function<
+    sdf::InterfaceModelPtr (sdf::NestedInclude include, Errors& errors)>;
+
+using sdf::StringPredicate = std::function<bool (const std::string)>;
```

Comment:
what type of string is passed to this function? is it the contents of the model file? it's not the file extension, right?

a doc-string for this type would help clarify things